symphony-ws/src/main/java/se/havochvatten/symphony/service/ReportService.java

```diff
@@ -602,151 +608,211 @@ public byte[] generateMultiComparisonAsODS(
+                nextSheet.getRange(ODF_TITLE_ROWS, 2 + layersToList.pressures.length * 2),
+                baseline_s, ODSStyles.singleSumSubHeader);
+            setCellValueAndStyle(
+                nextSheet.getRange(ODF_TITLE_ROWS, 2 + layersToList.pressures.length * 2 + 1),
+                diff_s, ODSStyles.singleSubHeader);
+
+            nextSheet.setColumnWidth(2 + layersToList.pressures.length * 2, ODF_TITLE_COLWIDTH);
+            nextSheet.setColumnWidth(2 + layersToList.pressures.length * 2 + 1, ODF_TITLE_COLWIDTH);
+
+            setCellValueAndStyle(
+                nextSheet.getRange(ODF_TITLE_ROWS + 1 + layersToList.ecosystems.length, 1),
+                metaDict.get("total").toUpperCase(), ODSStyles.totalHP);
 
             for (e = 0; e < layersToList.ecosystems.length; e++) {
-                nextCell = nextSheet.getRange(ODF_TITLE_ROWS + e, 2 + layersToList.pressures.length);
-                nextCell.setValue(cmp.totalPerEcosystem.get(cmp.includedEcosystems[layersToList.ecosystems[e]]));
-                nextCell.setStyle(ODSStyles.totalE);
+                setCellValueAndStyle(
+                    nextSheet.getRange(ODF_TITLE_ROWS + e + 1, 2 + layersToList.pressures.length * 2),
+                    cmp.totalPerEcosystem.get(cmp.includedEcosystems[layersToList.ecosystems[e]]).totalBaseline(), ODSStyles.totalE);
+                setCellValueAndStyle(
+                    nextSheet.getRange(ODF_TITLE_ROWS + e + 1, 2 + layersToList.pressures.length * 2 + 1),
+                    cmp.totalPerEcosystem.get(cmp.includedEcosystems[layersToList.ecosystems[e]]).totalDifference(), ODSStyles.totalE2);
             }
 
             for (p = 0; p < layersToList.pressures.length; p++) {
-                nextCell = nextSheet.getRange(ODF_TITLE_ROWS + layersToList.ecosystems.length, 2 + p);
-                nextCell.setValue(cmp.totalPerPressure.get(cmp.includedPressures[layersToList.pressures[p]]));
-                nextCell.setStyle(ODSStyles.totalP);
+                setCellValueAndStyle(
+                    nextSheet.getRange(ODF_TITLE_ROWS + layersToList.ecosystems.length + 1, 2 + p * 2),
+                    cmp.totalPerPressure.get(cmp.includedPressures[layersToList.pressures[p]]).totalBaseline(), ODSStyles.totalP);
+                setCellValueAndStyle(
+                    nextSheet.getRange(ODF_TITLE_ROWS + layersToList.ecosystems.length + 1, 2 + p * 2 + 1),
+                    cmp.totalPerPressure.get(cmp.includedPressures[layersToList.pressures[p]]).totalDifference(), ODSStyles.totalP);
             }
 
-            nextCell = nextSheet.getRange(
-                ODF_TITLE_ROWS + layersToList.ecosystems.length,
-                2 + layersToList.pressures.length);
-            nextCell.setValue(cmp.cumulativeTotal);
-            nextCell.setStyle(ODSStyles.totalC);
+            setCellValueAndStyle(
+                nextSheet.getRange(
+                    ODF_TITLE_ROWS + layersToList.ecosystems.length + 1,
```

Comment:
Maybe set a name for the numbers, it is kind of hard to get why we suddenly take `+ 1` here for example. It might also be easier not to miss-write an offset if there is a name related to the sheet/document.

Maybe it is too much but I just feel like it is a lot of integers everywhere which might be hard to keep track of

Reply:
So the + 1 in this case is for the additional subheader row just below the column titles: specifying "Baseline"/"Difference" (example in Swedish below)
![image](https://github.com/havochvatten/MSP-Symphony/assets/112889179/eea24409-57eb-4275-94ee-6be638ba4c68)
which is introduced to the report by this feature.

The `ODF_TITLE_ROWS` constant represents the offset from the top to the start of the table. Then we get the subheader row ( + 1 ) and the list of ecosystems follow vertically.
To ease readability I've tried to adhere to a convention where the numbers and constants appear at the "right place" in the expression. Note that I happen to deviate from that here, the `+ 1` should appear after the constant.

To make this specific instance clearer I see two options, either a new constant for the header offset including subheading
```
ODF_TITLE_ROWS = 8;
ODF_TITLE_ROWS_WITH_SUBHEADING = 9;
```
or to apply the previously mentioned "convention":  `(ODF_TITLE_ROWS + 1) + layersToList.ecosystems.length`.

🤔  I think I see what you're getting at. In this instance it doesn't really explain itself at a glance.
But I do think it'd become less readable to use constants "everywhere".

For example:
https://github.com/havochvatten/MSP-Symphony/blob/188441e21d0cc42921de6699486dd6dc9866e989/symphony-ws/src/main/java/se/havochvatten/symphony/service/ReportService.java#L718-L723
the consecutive rows here (+ 1, +2) does serve the readability better than constant variable names.

Reply:
Yes I agree that is could be too much and is good to keep like your example with `calcSectionOffset + 1` and `calcSectionOffset + 2` etc for consecutive rows.

I appreciate your fix with the other constants.